docker-entrypoint.sh

```diff
@@ -18,7 +18,11 @@ _ip_address() {
 	# scrape the first non-localhost IP address of the container
 	# in Swarm Mode, we often get two IPs -- the container IP, and the (shared) VIP, and the container IP should always be first
 	ip address | awk '
-		$1 == "inet" && $NF != "lo" {
+		$1 != "inet" { next } # only lines with ip addresses
+		$NF == "lo" { next } # skip loopback devices
+		$2 ~ /^127[.]/ { next } # skip loopback addresses
+		$2 ~ /^169[.]254[.]/ { next } # skip link-local addresses
```

Comment:
I suppose it's _technically_ valid for two containers to talk to each other using link-local addresses (or really a network/IPAM plugin to use link-local addresses for giving containers IP addresses) -- do we know whether there are any popular network/IPAM plugins that use link-local addresses?

`docker network connect` has `--link-local-ip` for adding an explicit link-local address to a container, but I'm not convinced that's very common.

Reply:
Apparently, `link-local` is not supported in docker-compose file v3 but was available in v2 (https://forums.docker.com/t/deprecation-of-link-local-ip/37342)